share/functions/__fish_prepend_sudo.fish

```diff
@@ -1,9 +1,17 @@
 function __fish_prepend_sudo -d "Prepend 'sudo ' to the beginning of the current commandline"
-    set -l cmd (commandline -poc)
+    set -l cmd (commandline -po)
+    set -l cursor (commandline -C)
     if test "$cmd[1]" != sudo
-        set -l cursor (commandline -C)
         commandline -C 0
         commandline -i "sudo "
         commandline -C (math $cursor + 5)
+    else
+        commandline -r (string sub --start=6 (commandline -p))
+        set -l new_cursor (math $cursor - 5)
```

Comment:
You can do the same with `commandline -C -- (math $cursor - 5)`

Reply:
Ah I can't believe I didn't think of this... any negative result of the math is being counted as an option to commandline that doesn't exist. Nice catch.